### PR TITLE
chore/compliance with js override

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose
richie-education 2.0.0-beta.15 introduces the possibility to override react components. This means that we can now write js code in richie-site-factory/sites so we need to be able to lint it and compile it.

## Proposal
- [x] Add js lint commands

- [x] Add js build commands

- [x] Upgrade CI conf

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -55,3 +56,12 @@ yarn-error.log
...
 .circleci/src/workflows/@*.yml
+
+### VisualStudioCode ###
+.vscode/*
+*.code-workspace
+
+### VisualStudioCode Patch ###
+# Ignore all local history of files
+.history
+.ionide
```
Comment (sampaccoud) [inline .gitignore:67]: All these are not related to the project and should be in the .gitignore of your own profile in your laptop.

File: Makefile [hunk 2]
```diff
@@ -104,6 +104,14 @@ build-sass-production: ## build Sass files to CSS (production mode)
 	@$(YARN) sass-production
 .PHONY: build-sass-production
 
+build-ts: ## build ts(x) files to JS
+	@$(YARN) build
+.PHONY: build-ts
+
+build-ts-production: ## build ts(x) files to js (production mode)
+	@$(YARN) build
```
Comment (sampaccoud) [inline Makefile:112]: did you forget production mode?

File: Makefile [hunk 3]
```diff
@@ -112,14 +120,24 @@ install-front-production: ## install front-end dependencies (production mode)
...
-lint-front-prettier-write: ## run prettier over scss files -- beware! overwrites files
+lint-front-prettier-write: ## run prettier over ts(x) & scss files -- beware! overwrites files
 	@$(YARN) prettier-write
 .PHONY: lint-front-prettier-write
 
+lint-front-eslint: ## run eslint over ts files
+	@$(YARN) lint
+.PHONY: lint-front-eslint
+
+lint-front: ## run frontend sources
```
Comment (sampaccoud) [inline Makefile:135]:
```suggestion
lint-front: ## run all linters frontend sources
```

File: bin/ci [hunk 4]
```diff
@@ -86,7 +143,23 @@ function update() {
 #
 # Usage: check_configuration
```
Comment (sampaccoud) [inline bin/ci:144]:
```suggestion
# Usage: check_configuration [-i | --ignore-changelog]
```

File: bin/ci [hunk 5]
```diff
@@ -68,15 +100,40 @@ function update_sources() {
 #
 # Usage: update
```
Comment (sampaccoud) [inline bin/ci:101]:
```suggestion
# Usage: update [-h | --help] [ -i | ignore-changelog]
```

File: bin/ci [hunk 6]
```diff
@@ -68,15 +100,40 @@ function update_sources() {
...
+for each site and running only the necessary jobs"
+    ignore_changelog=false
+
+    print_usage() {
+    printf "Usage: ..."
+    }
+
+    while true; do
+        case "${1}" in
+            --ignore-changelog|ignore-changelog)
```
Comment (sampaccoud) [inline bin/ci:113]: -i for a one letter option?

File: bin/ci [hunk 7]
```diff
@@ -68,15 +100,40 @@ function update_sources() {
 #
 # Usage: update
 function update() {
-    update_sources
+    help="Utility function that generates a circle configuration file with a workflow
+for each site and running only the necessary jobs"
+    ignore_changelog=false
+
+    print_usage() {
```
Comment (sampaccoud) [inline bin/ci:107]: copy/pasta fail

File: bin/ci [hunk 8]
```diff
@@ -68,15 +100,40 @@ function update_sources() {
...
+                usage_function "${FUNCNAME[0]}" "$help"
+                exit 0
+                ;;
+            *)
+                break
+                ;;
+        esac
+    done
+
+    update_factory ${ignore_changelog}
```
Comment (sampaccoud) [inline bin/ci:127]:
```suggestion
    update_factory "${ignore_changelog}"
```
shellcheck should correct all this

File: bin/ci (hunk 6 above)
Comment (jbpenrath) [reply bin/ci:113]: I hesitate because this shorcuted option did not mention changelog.

Comment (sampaccoud) [reply bin/ci:113]: well... it's just a letter :man_shrugging: could be anything in the alphabet?

File: bin/ci [hunk 9]
```diff
@@ -56,37 +83,79 @@ function update_sources() {
             echo "${site}")
 
         # shellcheck disable=SC2016
-        sed 's|${SITE}|'"${site}"'|g; s|${IMAGE}|'"${image}"'|g' \
-            < "${template}" >> ".circleci/src/workflows/@${site}.yml"
+        sed_cmd="s|\${SITE}|$site|g;s|\${IMAGE}|$image|g;"
+        if [ "$1" = true ] ; then
+            sed_cmd="$sed_cmd/CHANGELOG/,/lint-changelog/{//!d;};"
```
Comment (sampaccoud) [inline bin/ci:88]: Can you add a comment in english to describe what this does?

File: bin/ci [hunk 10]
```diff
@@ -17,19 +17,46 @@ COMMAND:
...
+    fi
+
+    sed "s|\${CI-UPDATE-OPTIONS}|$flags|;" < $template >> ".circleci/src/workflows/site-factory.yml"
+}
 
 # Utility function that updates circle configuration source files with a
 # workflow for each site and running only the necessary jobs
 #
-# Usage: update_sources
+# Usage: update_sources [--ignore-changelog]
```
Comment (sampaccoud) [inline bin/ci:58]:
```suggestion
# Usage: update_sources [-i | --ignore-changelog | ignore-changelog]
```

File: bin/ci [hunk 11]
```diff
@@ -17,19 +17,46 @@ COMMAND:
...
+# --ignore-changelog flag
+# Usage: update_factory [--ignore-changelog]
+function update_factory() {
+    disclaimer="# /!\ DO NOT EDIT: this file is autogenerated"
+    template=".circleci/src/workflows/site-factory.yml.tpl"
+    
+    # Add a workflow for the site from our template
+    printf "%s\n" "${disclaimer}" > ".circleci/src/workflows/site-factory.yml"
+
+    flags=""
```
Comment (sampaccoud) [inline bin/ci:47]: options?

File: bin/ci [hunk 12]
```diff
@@ -17,19 +17,46 @@ COMMAND:
...
+
 
 function check_changes() {
     to="${CIRCLE_SHA1:-HEAD}"
     git whatchanged --name-only --pretty="" origin/master.."${to}" | grep "${1}" &> /dev/null
 }
 
+# Utility function that updates circle configuration site-factory file according to
+# --ignore-changelog flag
+# Usage: update_factory [--ignore-changelog]
```
Comment (sampaccoud) [inline bin/ci:39]:
```suggestion
# Usage: update_factory [-i | --ignore-changelog | ignore-changelog]
```

File: bin/ci [hunk 13]
```diff
@@ -17,19 +17,46 @@ COMMAND:
 "
 }
 
+function usage_function() {
+    echo -e "usage: ${0} ${1} [--ignore-changelog | ignorechangelog] [-h | --help | help]
```
Comment (sampaccoud) [inline bin/ci:21]:
```suggestion
    echo -e "usage: ${0} ${1} [-i | --ignore-changelog | ignorechangelog] [-h | --help | help]
```

Comment (jbpenrath) [reply bin/ci:21]: Good catch!

File: bin/ci (hunk 9 above)
Comment (sampaccoud) [reply bin/ci:88]: Or maybe it is clearer to add a variable on line 10 to define ignore conditions?
So you can use the same pattern as for the site and image : search the variable and replace it by ignore conditions?

File: bin/ci (hunk 10 above)
Comment (jbpenrath) [reply bin/ci:58]: Hmm in fact, `update_sources` just received a boolean value.

`# Usage: update_sources ignore-changelog<boolean>` ?

File: bin/ci (hunk 12 above)
Comment (jbpenrath) [reply bin/ci:39]: Ditto

File: bin/ci (hunk 10 above)
Comment (sampaccoud) [reply bin/ci:58]: true :thinking: ok!

File: bin/ci (hunk 9 above)
Comment (jbpenrath) [reply bin/ci:88]: The way I implemeted it is not robust at all.
So I wrapped check-changelog job between #!CHANGELOG_OPTION in template file
And if ignore-changelog is true, I delete all delete wrapped into these comments

**Review by sampaccoud — APPROVED** [review]
